docs/source/write_first_app.rst

```diff
@@ -1,1014 +1,572 @@
-      discovery: {enabled: true, asLocalhost: true} // using asLocalhost as this gateway is using a fabric network deployed locally
-    });
+Third, access the smart contract to be invoked
+----------------------------------------------
+The sample application uses the ``Gateway`` connection to get a reference to the ``Network`` and then the default
+``Contract`` within a chaincode deployed on that network.
 
-    // Build a network instance based on the channel where the smart contract is deployed
-    const network = await gateway.getNetwork(channelName);
+.. code-block:: TypeScript
 
+  const channelName = 'mychannel';
+  const chaincodeName = 'basic';
 
-    // Get the contract from the network.
-    const contract = network.getContract(chaincodeName);
+  const network = gateway.getNetwork(channelName);
+  const contract = network.getContract(chaincodeName);
 
-When a chaincode package includes multiple smart contracts, on the `getContract() API <https://hyperledger.github.io/fabric-sdk-node/release-2.2/module-fabric-network.Network.html#getContract>`__ you can specify both the name of the chaincode package and a specific smart contract to target. For example:
+When a chaincode package includes multiple smart contracts, you can provide both the name of the chaincode and the name
+of a specific smart contract as arguments to the `getContract() <https://hyperledger.github.io/fabric-gateway/main/api/node/interfaces/Network.html#getContract>`_
+call. For example:
 
-.. code:: bash
+.. code-block:: TypeScript
 
-  const contract = await network.getContract('chaincodeName', 'smartContractName');
+  const contract = network.getContract('chaincodeName', 'smartContractName');
 
-Fourth, the application initializes the ledger with some sample data
---------------------------------------------------------------------
 
-Now that we are at the point where we are actually having the sample application submit transactions, let’s
-go through them in sequence. The application code snippets and invoked chaincode snippets
-are provided for each called function, as well as the terminal output.
+Fourth, populate the ledger with sample assets
+----------------------------------------------
+Immediately after initial deployment of the chaincode package, the ledger is empty. The application uses
+``submitTransaction()`` to invoke the ``InitLedger`` transaction function, which populates the ledger with some sample
+data. ``submitTransaction()`` will use the Gateway to:
 
-The submitTransaction() function is used to invoke the chaincode ``InitLedger`` function to populate the
-ledger with some sample data. Under the covers, the submitTransaction() function will use service discovery
-to find a set of required endorsing peers for the chaincode, invoke the chaincode
-on the required number of peers, gather the chaincode endorsed results from those peers,
-and finally submit the transaction to the ordering service.
+  1. Endorse the transaction proposal.
```

Comment:
It was described at a high level previously but was deemed confusing, so avoided.

Reply:
@joshhus what was the concern? A sentence or two would definitely help here. Feel free to propose an edit to what I suggested.

Reply:
The discussion here was that it reads as if the "endorsement" by one peer / org then leads to a commitment to the ledger. Without a full understanding of endorsement and endorsement policies, this could be misleading to a reader. We discussed this in the other topics also - endorsement by one org does not mean the Tx passes endorsement (policies). So generally a semantics issue of being imprecise with the word "endorsed" in different contexts.

Reply:
Actually the comments before were on whether the brief description of executing the transaction on peers matched the description of the endorsement process we already have in other pages.

My intent with this list of steps is to signpost the following application flow, which is what gets executed when you call `submitTransaction()`, not to describe the transaction flow at the server side:
- [endorse()](https://hyperledger.github.io/fabric-gateway/main/api/node/interfaces/Proposal.html#endorse)
- [submit()](https://hyperledger.github.io/fabric-gateway/main/api/node/interfaces/Transaction.html#submit)
- [getStatus()](https://hyperledger.github.io/fabric-gateway/main/api/node/interfaces/SubmittedTransaction.html#getStatus) to wait for the transaction commit

Understanding the application API flow is important for interpreting error types that may be generated by that flow, touched on in the last step.